zwave_js_server/model/node/__init__.py

```diff
@@ -366,8 +366,14 @@ def update(self, data: NodeDataType) -> None:
         for value_id in stale_value_ids:
             self.values.pop(value_id)
 
-        # Updating existing values and populate new values
-        for value_id in new_value_ids - stale_value_ids:
+        # Updating existing values and populate new values. Preserve value order if
```

Comment:
Why does the order matter?

Reply:
Not sure yet but without this change existing HA tests were failing presumably because it impacted the way entities got named and the same state calls from before were getting different values.

I opened a TODO issue so I don't lose track of this, but for now this will avoid a breaking change. The proper fix may end up being a breaking change on the HA side